src/dht_proxy_server.cpp

```diff
@@ -129,6 +132,13 @@ DhtProxyServer::DhtProxyServer(std::shared_ptr<DhtRunner> dht, in_port_t port ,
                 scheduler_.edit(printStatsJob_, scheduler_.time() + PRINT_STATS_PERIOD);
             }
         });
+        getNodeStatsJob_ = scheduler_.add(scheduler_.time(), [this]{
+            if (service_->is_up() and not stopListeners) {
+                ipv4Stats_ = dht_->getNodesStats(AF_INET);
```

Comment:
not thread-safe but I guess that's ok since it's a POD and consistency is not a requirement for this data.

Reply:
dht_ should exists till the proxy exists. And the scheduler can only launch this job once at a time?

Reply:
dht_ is not the issue, consistency of ipv4Stats_ is but it doean't really matter

Reply:
Yeah the only issue I see is that we make a GET / at the same time the scheduler is called. Here we will have 2 write operations for the same value. In the other case the scheduler can't start two `getNodeStatsJob_` at the same time?

Anyway if we decide to keep the write operations in `getNodeInfo` I can add a lock_guard to be sure.

Reply:
There also might be one or more read operations while data is written so the data might be inconsistent.

Reply:
arf yeah I forgot that the write operation is not atomic :/

I'll add a lock to be sure

Reply:
Done